### PR TITLE
refactoring the index and sidebar

### PR DESCRIPTION
**Pull request by choldgraf**
Attempt at refactoring the landing page for the Binder docs, using some information from the repo2docker documentation, and also reorganizing some things on the main index page so that the sidebar is broken up by section now.

cc @betatim and @yuvipanda who were on the PR in repo2docker where I mentioned doing this before.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+version: 2
+jobs:
+  build_docs:
+    docker:
+      - image: circleci/python:3.6-stretch
+    steps:
+      # Get our data and merge with upstream
+      - run: sudo apt-get update
+      - checkout
+      - run: echo $(git log -1 --pretty=%B) | tee gitlog.txt
```
Comment (betatim) [inline .circleci/config.yml]: does circleci really need all the steps between L10 and L21 just to build a python package? Feels like boilerplate hell :-/

Comment (choldgraf) [reply .circleci/config.yml]: no idea haha I'm still debugging that

Comment (choldgraf) [reply .circleci/config.yml]: (this is copypasta from another repo...I'm trying to figure out how to get the artifacts to load properly)

**Comment by choldgraf** [comment]
got circle working! here are the built docs:

https://14-100564515-gh.circle-artifacts.com/0/html/index.html

LMK what folks think

**Comment by betatim** [comment]
screen shot 2018-09-24 at 07 38 11

The word "Advanced" is missing in the sidebar.

How about moving the FAQ up into "Getting started with Binder" as a third bullet point (same level as the current two bullet points in that section)?

**Comment by choldgraf** [comment]
I guess we could re-arrange the categories, though I don't feel that it's much of an improvement over the current structure (e.g. I like having "how to" as a separate section so that we can list how-to items underneath it...I assume users don't usually click-through beyond the first page). 

re: making Binder documentation independent of `mybinder.org`, I think in general this is a good idea though perhaps we should take discussion of this into an issue rather than in this PR :-)

**Comment by betatim** [comment]
Sounds good. I was wondering if we could make it easier to get orientated by using sections people would recognise from other docs pages.

**Comment by choldgraf** [comment]
OK, docs are building fine now! I think we could still improve the organization etc a little bit, but I'd prefer if we could do this via another PR or discussion in issues, since I think this PR makes a tangible benefit to what is currently on the Binder docs. What do folks think?

**Pull request merged**